kclvm/sema/src/resolver/doc.rs

```diff
@@ -274,19 +274,29 @@ pub(crate) fn parse_doc_string(ori: &String) -> Doc {
 
     let attrs = parse_attr_list(attr_content);
 
-    Doc::new(summary, attrs)
+    let example_section = read_to_next_section(&mut doc);
+    let default_example_content = match example_section.len() {
+        0 | 1 | 2 => "".to_string(),
+        _ => example_section[2..].join("\n"),
+    };
+    let mut examples = HashMap::new();
+    examples.insert("Default example".to_string(), Example::new("".to_string(), "".to_string(), default_example_content));
+
+
+    Doc::new(summary, attrs, examples)
 }
 
 /// The Doc struct contains a summary of schema and all the attributes described in the the docstring.
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub(crate) struct Doc {
     pub summary: String,
     pub attrs: Vec<Attribute>,
+    pub examples: HashMap<String, Example>,
 }
 
 impl Doc {
-    fn new(summary: String, attrs: Vec<Attribute>) -> Self {
-        Self { summary, attrs }
+    fn new(summary: String, attrs: Vec<Attribute>, examples: HashMap<String, Example>) -> Self {
+        Self { summary, attrs, examples: examples }
```

Comment:
Just write the code. `Self { summary, attrs, examples }`

Reply:
fixed, ptal